x-pack/plugins/enterprise_search/public/applications/shared/query_params/query_params.ts

```diff
@@ -7,5 +7,17 @@
 
 import queryString from 'query-string';
 
-export const parseQueryParams = (search: string) =>
-  queryString.parse(search, { arrayFormat: 'bracket' });
+const PARSE_OPTIONS: queryString.ParseOptions = { arrayFormat: 'bracket' };
+const STRINGIFY_OPTIONS: queryString.StringifyOptions = {
+  arrayFormat: 'bracket',
+  encode: true,
+  strict: true,
+};
+
+export const parseQueryParams = (search: string) => queryString.parse(search, PARSE_OPTIONS);
+
+export const addQueryParameter = (url: string, key: string, value: string) => {
```

Comment:
suggestion: we probably want to use generateEncodedPath to construct these URLs, that'll take care of some nasty edge cases.

Reply:
One of the reasons I added a function here is to use the `query-string` library that is handling encoding etc.

it looks like `generateEncodedPath` will do what I need without adding a new function, so I'll update to use that instead.